src/methods/clipping/intersection.jl

```diff
@@ -0,0 +1,423 @@
+# #  Intersection Polygon Clipping
+export intersection, intersection_points
+
+# This 'intersection' implementation returns the intersection of two polygons.
+# It returns a Vector{Vector{Vector{Tuple{Float}}}. The Vector{Vector{Tuple{Float}
+# is empty if the two polygons don't intersect. The algorithm to determine the 
+# intersection was adapted from "Efficient clipping of efficient polygons," by 
+# Greiner and Hormann (1998). DOI: https://doi.org/10.1145/274363.274364
+
+"""
+    intersection(geom_a, geom_b)::Union{Tuple{::Real, ::Real}, ::Nothing}
+
+Return an intersection point between two geometries. Return nothing if none are
+found. Else, the return type depends on the input. It will be a union between:
+a point, a line, a linear ring, a polygon, or a multipolygon
+
+## Example
+
+```jldoctest
+import GeoInterface as GI, GeometryOps as GO
+
+line1 = GI.Line([(124.584961,-12.768946), (126.738281,-17.224758)])
+line2 = GI.Line([(123.354492,-15.961329), (127.22168,-14.008696)])
+GO.intersection(line1, line2)
+
+# output
+(125.58375366067547, -14.83572303404496)
+```
+"""
+intersection(geom_a, geom_b) =
+    intersection(GI.trait(geom_a), geom_a, GI.trait(geom_b), geom_b)
+"""
+    intersection(
+        ::GI.PolygonTrait, poly_a,
+        ::GI.PolygonTrait, poly_b,
+    )::Vector{Vector{Vector{Tuple{Float64}}}}    
+
+Calculates the intersection between two polygons. If the intersection is empty, 
+the vector of a vector is empty (note the outermost vector is technically not empty).
+## Example
+
+```jldoctest
+import GeoInterface as GI, GeometryOps as GO
+
+p1 = GI.Polygon([[(0.0, 0.0), (5.0, 5.0), (10.0, 0.0), (5.0, -5.0), (0.0, 0.0)]])
+p2 = GI.Polygon([[(3.0, 0.0), (8.0, 5.0), (13.0, 0.0), (8.0, -5.0), (3.0, 0.0)]])
+GO.intersection(p1, p2)
+
+# output
+1-element Vector{Vector{Vector{Tuple{Float64, Float64}}}}:
+[[[(6.5, 3.5), (10.0, 0.0), (6.5, -3.5), (3.0, 0.0), (6.5, 3.5)]]]
+```
+"""
+
+function intersection(::GI.PolygonTrait, poly_a, ::GI.PolygonTrait, poly_b)
+    # First we get the exteriors of 'poly_a' and 'poly_b'
+    ext_poly_a = GI.getexterior(poly_a)
```

Comment:
Since we only need to pass rings (exterior or holes) to `_build_ab_list`, we should just pass  `GI.getexterior(poly_a)` and `GI.getexterior(poly_b)` in and then make any changes to `_build_ab_list` (hopefully none) to make that work. Then you don't need to make the polygons since we don't need to dispatch within our helper function.